docs/architecture/accounts.md

```diff
@@ -1,211 +1,139 @@
-  #!
-  #! ...
-  export.distribute
-      # get max supply of this faucet. We assume it is stored at pos 3 of slot 1
-      push.METADATA_SLOT exec.account::get_item drop drop drop
-      # => [max_supply, amount, tag, note_type, RECIPIENT, ...]
+This encoding allows the ID to convey both the account’s unique identity and its operational settings.
 
-      # get total issuance of this faucet so far and add amount to be minted
-      exec.faucet::get_total_issuance
-      # => [total_issuance, max_supply, amount, tag, note_type RECIPIENT, ...]
+### Storage
 
-      # compute maximum amount that can be minted, max_mint_amount = max_supply - total_issuance
-      sub
-      # => [max_supply - total_issuance, amount, tag, note_type, RECIPIENT, ...]
+> A flexible, arbitrary data store within the `Account`.
 
-      # check that amount <= max_supply - total_issuance, fails if otherwise
-      dup.1 gte assert.err=ERR_BASIC_FUNGIBLE_MAX_SUPPLY_OVERFLOW
-      # => [asset, tag, note_type, RECIPIENT, ...]
+The [storage](../../objects/src/accounts/storage/mod.rs) of an `Account` consists of up to 255 indexed [storage slots](../../objects/src/accounts/storage/slot/mod.rs). Each slot can be one of the following types:
 
-      # creating the asset
-      exec.asset::create_fungible_asset
-      # => [ASSET, tag, note_type, RECIPIENT, ...]
+- **`StorageSlot::Value`:** Contains a single `Word` (32 bytes) of arbitrary data.  
+- **`StorageSlot::Map`:** Contains a [StorageMap](../../objects/src/accounts/storage/map.rs), a key-value store where both keys and values are `Word`s. The slot's value is a commitment (hash) to the entire map.
 
-      # mint the asset; this is needed to satisfy asset preservation logic.
-      exec.faucet::mint
-      # => [ASSET, tag, note_type, RECIPIENT, ...]
+### Nonce
 
-      # store and drop the ASSET
-      mem_storew.3 dropw
-      # => [tag, note_type, RECIPIENT, ...]
+> A counter incremented with each state update to the `Account`.
 
-      # create a note containing the asset
-      exec.tx::create_note
-      # => [note_ptr, ZERO, ZERO, ...]
+The `nonce` enforces ordering and prevents replay attacks or double-spending. It must strictly increase with every account state update. The increment must be less than `2^32` but always greater than the previous nonce, ensuring a well-defined sequence of state changes.
 
-      # store and drop the ASSET
-      padw mem_loadw.3 movup.4 exec.tx::add_asset_to_note
-      # => [note_ptr, ASSET, ZERO, ...]
-  end
+### Vault
 
-  #! Burns fungible assets.
-  #!
-  #! ...
-  export.burn
-      # burning the asset
-      exec.faucet::burn
-      # => [ASSET]
+> A collection of [assets](assets.md) stored by the `Account`.
 
-      # increments the nonce (anyone should be able to call that function)
-      push.1 exec.account::incr_nonce
+Assets are stored in a sparse Merkle tree, allowing for efficient, cryptographically secure proofs of ownership and state:
```

Comment:
Do we need to explain that they are stored in a sparse Merkle tree? 

I know I also had it in my version, but maybe its better to stick to the protocol and not the implementation. 

So we could say how big the vault is and that it can hold fungible and non-fungible assets.

Reply:
Changed it to something shorter and less implementation focused.